### PR TITLE
INTRN-218: Rewrite logger; notably handling of uvicorn loggers and exception formatting

### PR DESCRIPTION
**Pull request by cmyui**
this is super annoying; look at how useless this `exc_info` is: https://app.datadoghq.com/logs?query=status%3Aerror%20service%3Ainterns-backend&cols=host%2Cservice&event=AgAAAYg71Zcly_8c2AAAAAAAAAAYAAAAAEFZZzcxYUdLQUFCcHg4RVNkQmlHaGdBTgAAACQAAAAAMDE4ODNiZDUtYzNjMC00OTY1LWFjNWMtZTBiM2Q5YTE5ZTE4&index=%2A&messageDisplay=inline&stream_sort=desc&viz=stream&from_ts=1680750604418&to_ts=1680751504418&live=true

### CODE REVIEW

File: .github/workflows/staging-deploy.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - INTRN-218-fix-json-stacktrace-logging
```
Comment (tsunyoku) [inline .github/workflows/staging-deploy.yml]: remember to remove this lol

File: app/logger.py [hunk 2]
```diff
@@ -27,14 +28,7 @@ def get_logger(name: str | None = None) -> structlog.stdlib.BoundLogger:
...
 def log_as_text(app_env: str) -> bool:
-    # TODO: fix error with json logger where it does this with stacktraces
-    # https://app.datadoghq.com/logs?query=status%3Aerror%20service%3Ainterns-backend&cols=host%2Cservice&event=AgAAAYg71Zcly_8c2AAAAAAAAAAYAAAAAEFZZzcxYUdLQUFCcHg4RVNkQmlHaGdBTgAAACQAAAAAMDE4ODNiZDUtYzNjMC00OTY1LWFjNWMtZTBiM2Q5YTE5ZTE4&index=%2A&messageDisplay=inline&stream_sort=desc&viz=stream&from_ts=1680750604418&to_ts=1680751504418&live=true
-    # return app_env == "local"
-    return True
-
-
-def log_with_colors(app_env: str) -> bool:
-    return app_env == "local"
+    return False
```
Comment (tsunyoku) [inline app/logger.py]: revert this back to the app_env check

Comment (cmyui) [inline app/logger.py]:
```suggestion
    return app_env == "local"
```

File: app/logger.py [hunk 3]
```diff
@@ -49,38 +48,97 @@ def add_request_id(_: WrappedLogger, __: str, event_dict: EventDict) -> EventDic
...
+        stdlib_logging.getLogger(_logger).handlers.clear()
+        stdlib_logging.getLogger(_logger).propagate = True
 
-        # defer logging control to the root logger
-        logger.propagate = True
-        logger.setLevel(log_level)
-        for logger_handler in logger.handlers:
-            logger.removeHandler(logger_handler)
+    # effectively disable uvicorn.access; we will recreate this via middleware
+    stdlib_logging.getLogger("uvicorn.access").handlers.clear()
```
Comment (cmyui) [inline app/logger.py]:
```suggestion
    # effectively disable uvicorn.access
    # TODO: recreate access logs using middleware
    stdlib_logging.getLogger("uvicorn.access").handlers.clear()
```

File: app/logger.py [hunk 4]
```diff
@@ -49,38 +48,98 @@ def add_request_id(_: WrappedLogger, __: str, event_dict: EventDict) -> EventDic
...
-        logger.setLevel(log_level)
-        for logger_handler in logger.handlers:
-            logger.removeHandler(logger_handler)
+    # effectively disable uvicorn.access
+    # TODO: recreate access logs using middleware
+    stdlib_logging.getLogger("uvicorn.access").handlers.clear()
+    stdlib_logging.getLogger("uvicorn.access").propagate = False
+
+    uvicorn_access_logger = stdlib_logging.getLogger("uvicorn")
+    uvicorn_access_logger.addHandler(handler)
```
Comment (cmyui) [inline app/logger.py]:
```suggestion
```

File: .github/workflows/staging-deploy.yml (hunk 1 above)
Comment (cmyui) [reply .github/workflows/staging-deploy.yml]:
```suggestion
```

**Review by tsunyoku — APPROVED** [review]

**Pull request merged**